esp/src/src-react/components/Monitoring.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from "react";
+import { CommandBar, ContextualMenuItemType, ICommandBarItemProps, Image, Link } from "@fluentui/react";
+import { MachineService } from "@hpcc-js/comms";
+import * as domClass from "dojo/dom-class";
+import { ShortVerticalDivider } from "./Common";
+import { tree } from "./DojoGrid";
+import { useFluentGrid } from "../hooks/grid";
+import { HolyGrail } from "../layouts/HolyGrail";
+import * as Utility from "src/Utility";
+import nlsHPCC from "src/nlsHPCC";
+
+function getStatusImageName(row) {
+    switch (row.Status) {
+        case "Normal":
+        default:
+            return "normal.png";
+        case "Warning":
+            return "warning.png";
+        case "Error":
+            return "error.png";
+    }
+}
+
+const machine = new MachineService({ baseUrl: "" });
+
+const defaultUIState = {
+    hasSelection: false
+};
+
+interface MonitoringProps {
+
+}
+
+export const Monitoring: React.FunctionComponent<MonitoringProps> = ({
+
+}) => {
+
+    const [uiState, setUIState] = React.useState({ ...defaultUIState });
+    const [data, setData] = React.useState<any[]>([]);
+
+    //  Grid ---
+    const { Grid, selection, copyButtons } = useFluentGrid({
+        data,
+        primaryID: "__hpcc_id",
+        sort: { attribute: "Name", "descending": false },
+        filename: "monitoring",
+        columns: {
+            StatusID: { label: "", width: 0, sortable: false, hidden: true },
+            ComponentType: tree({
+                label: "Name", sortable: true, width: 200,
+                formatter: function (Name, row) {
+                    return <Image src={Utility.getImageURL(getStatusImageName(row))} />;
+                }
+            }),
+            StatusDetails: { label: "Details", sortable: false },
+            URL: {
+                label: "URL", width: 200, sortable: false,
+                formatter: function (Name, row) {
+                    if (Name) {
+                        return <Link href={`http://${Name}`} target="_blank">{Name}</Link>;
+                    } else {
+                        return "";
+                    }
+                }
+            },
+            EndPoint: { label: "IP", sortable: true, width: 140 },
+            TimeReportedStr: { label: "Time Reported", width: 140, sortable: true },
+            Status: {
+                label: nlsHPCC.Severity, width: 130, sortable: false,
+                formatter: function (object, value, node, options) {
+                    switch (value) {
+                        case "Error":
+                            domClass.add(node, "ErrorCell");
```

Comment:
Should probably not use "domClass", also I am not 100% sure you will even get a "node" here?